src/Microsoft.ML.Featurizers/DateTimeTransformer.cs

```diff
@@ -455,11 +465,17 @@ internal abstract class TypedColumn : IDisposable
         {
             internal readonly string Source;
             internal readonly string Prefix;
+            internal readonly int IntPtrSize;
+            internal readonly int StructSize;
 
             internal TypedColumn(string source, string prefix)
             {
                 Source = source;
                 Prefix = prefix;
+                IntPtrSize = IntPtr.Size;
+
+                // The native struct is 25 bytes + 8 size_t.
+                StructSize = 25 + (IntPtrSize * 8);
```

Comment:
Is it possible to avoid hardcoded numbers? Would Marshal.Sizeof on the native struct work?

Reply:
If I remember correctly I tried that and had issues with it, but let me try again.

Reply:
Oh, the issue was that the native struct returns pointers to a string buffer and my struct uses strings directly. I can use Marshal.SizeOf, but then I need to define a second struct that exactly matches the native code. Would you prefer that approach?

Reply:
Yes, I would prefer that approach. The presence of a struct that matches the native code makes the contract explicit and if something were to change on the native side, the corresponding change would be easier to identify.

---
In reply to: [368129495](https://github.com/dotnet/machinelearning/pull/4661#discussion_r368129495) [](ancestors = 368129495)

Reply:
Sounds good. I have added in another struct that exactly matches the native code and am using that to get the size of the struct.